module/zfs/abd.c

```diff
@@ -102,6 +102,50 @@
 #include <sys/zfs_znode.h>
 #include <sys/debug.h>
 
+#ifdef DEBUG
+#ifdef _KERNEL
+#define VERIFY_ABD_MAGIC(x) do {	      \
+		const uint64_t y = (x)->abd_magic;	\
+		if (y != ABD_DEBUG_MAGIC) {		\
+			panic("VERIFY_ABD_MAGIC(" #x ") failed (0x%llx != 0x%llx )\n", \
+			    y, ABD_DEBUG_MAGIC);			\
+		}							\
+	} while (0)
+#else
+#define VERIFY_ABD_MAGIC(x) do {	     \
+	  const uint64_t y = (x)->abd_magic; \
+	  if (y != ABD_DEBUG_MAGIC) {		     \
+		  char * __buf = alloca(256);				\
+		  (void) snprintf(__buf, 256,				\
+		      "VERIFY_ABD_MAGIC(%s) failed (0x%llx != 0x%llx)", \
+		      #x, y, ABD_DEBUG_MAGIC);				\
+		  __assert_c99(__buf, __FILE__, __LINE__, __func__);	\
+	  }								\
+	} while (0)
+#endif
+#endif
+
+#ifdef DEBUG
+#ifdef _KERNEL
+#define VERIFY_BUF_NOMAGIC(x, size) do {				\
+		const uint64_t m = ((abd_t *)(x))->abd_magic;		\
+		if ((size) >= sizeof(abd_t) && m == ABD_DEBUG_MAGIC) {	\
+			panic("VERIFY_BUF_NOMAGIC(" #x ", 0x%lx) failed\n", size); \
+		}							\
+	} while (0)
+#else
+#define VERIFY_BUF_NOMAGIC(x, size) do {				\
+		const uint64_t m = ((abd_t *)(x))->abd_magic;		\
+		if ((size) >= sizeof(abd_t) && m == ABD_DEBUG_MAGIC) {	\
+			char *__buf = alloca(256);			\
+			(void) snprintf(__buf, 256,			\
+			    "VERIFY_BUF_NOMAGIC(%s, 0x%lx)) failed", #x, size); \
+			__assert_c99(__buf, __FILE__, __LINE__, __func__); \
+		}							\
+	} while (0)
+#endif
+#endif
+
 typedef struct abd_stats {
```

Comment:
Where do we define `VERIFY_BUF_NOMAGIC` when compiling without DEBUG ?

Reply:
d'oh!  sorry!